allow non used values to be preserved when templating a url

Jmeyering opened the pull request:
This PR implements the suggestion mentioned in #3 .

However I don't have tests implemented yet, and this needs to be toggled via an options, rather than working by default.

Let me know if this is the best way to go about this, and how implementing this as an expansion options would work.

pleonasm commented:
Ah yes, this functionality would break things if it was on by default. Adding a third optional parameter to __invoke() on the Expander to allow for the functionality to be switched on or off would be fine (defaulting to the old behavior, of course).

Jmeyering commented:
Okay I added a third param in the form of an options array.

The option `preserveTpl => true` Will trigger the template preservation.

Let me know about the test suite. I didn't add a lot of tests but I added tests that cover my particular use cases. Let me know about that.

Jmeyering commented:
Just wondering if you have had a chance to look into this PR. Thanks!
